diff -Naur bb.org-overlays.orig/Makefile bb.org-overlays/Makefile
--- bb.org-overlays.orig/Makefile	2019-04-04 18:33:29.000000000 +0200
+++ bb.org-overlays/Makefile	2019-04-25 10:15:14.863379567 +0200
@@ -1,6 +1,7 @@
 
 DTC ?= /usr/bin/dtc
 CPP ?= cpp
+CC ?= gcc
 DESTDIR ?=
 
 MAKEFLAGS += -rR --no-print-directory
@@ -87,7 +88,7 @@
 
 all_%:
 	$(Q)$(MAKE) ARCH=$* all_arch
-	gcc -o config-pin ./tools/pmunts_muntsos/config-pin.c
+	$(CC) -o config-pin ./tools/pmunts_muntsos/config-pin.c
 
 clean_%:
 	$(Q)$(MAKE) ARCH=$* clean_arch
