diff -Naur bborg-overlays.orig/src/arm/BB-BONE-LCD7-01-00A1.dts bborg-overlays/src/arm/BB-BONE-LCD7-01-00A1.dts
--- bborg-overlays.orig/src/arm/BB-BONE-LCD7-01-00A1.dts	2019-06-20 12:00:49.335952059 +0200
+++ bborg-overlays/src/arm/BB-BONE-LCD7-01-00A1.dts	2019-06-20 12:03:59.942211672 +0200
@@ -210,43 +210,33 @@
 
 				button@1 {
 					debounce_interval = <50>;
-					linux,code = <105>;
-					label = "left";
+					linux,code = <59>;
+					label = "f1";
 					gpios = <&gpio1 16 0x1>;
-					gpio-key,wakeup;
-					autorepeat;
 				};
 				button@2 {
 					debounce_interval = <50>;
-					linux,code = <106>;
-					label = "right";
+					linux,code = <60>;
+					label = "f2";
 					gpios = <&gpio1 17 0x1>;
-					gpio-key,wakeup;
-					autorepeat;
 				};
 				button@3 {
 					debounce_interval = <50>;
-					linux,code = <103>;
-					label = "up";
+					linux,code = <61>;
+					label = "f3";
 					gpios = <&gpio1 19 0x1>;
-					gpio-key,wakeup;
-					autorepeat;
 				};
 				button@4 {
 					debounce_interval = <50>;
-					linux,code = <108>;
-					label = "down";
+					linux,code = <62>;
+					label = "f4";
 					gpios = <&gpio3 16 0x1>;
-					gpio-key,wakeup;
-					autorepeat;
 				};
 				button@5 {
 					debounce_interval = <50>;
-					linux,code = <28>;
-					label = "enter";
+					linux,code = <63>;
+					label = "f5";
 					gpios = <&gpio1 12 0x1>;
-					gpio-key,wakeup;
-					autorepeat;
 				};
 			};
 
