diff -Naur xcsoar-v6.8.11.orig/src/Event/Poll/Linux/Input.cpp xcsoar-v6.8.11/src/Event/Poll/Linux/Input.cpp
--- xcsoar-v6.8.11.orig/src/Event/Poll/Linux/Input.cpp	2018-08-18 16:36:49.000000000 +0200
+++ xcsoar-v6.8.11/src/Event/Poll/Linux/Input.cpp	2019-04-26 09:17:59.987623719 +0200
@@ -33,6 +33,7 @@
 
 #include <termios.h>
 #include <sys/ioctl.h>
+#include <errno.h>
 
 template<typename T>
 static constexpr unsigned
