diff -Naur xcsoar-v6.8.11.orig/build/options.mk xcsoar-v6.8.11/build/options.mk
--- xcsoar-v6.8.11.orig/build/options.mk	2018-08-18 16:36:49.000000000 +0200
+++ xcsoar-v6.8.11/build/options.mk	2019-05-15 19:57:31.538439293 +0200
@@ -62,3 +62,7 @@
 ifneq ($(WGS84),n)
 TARGET_CPPFLAGS += -DUSE_WGS84
 endif
+
+ifdef FIXED_DATADIR
+TARGET_CPPFLAGS += -DFIXED_DATADIR=\"$(FIXED_DATADIR)\"
+endif
diff -Naur xcsoar-v6.8.11.orig/src/LocalPath.cpp xcsoar-v6.8.11/src/LocalPath.cpp
--- xcsoar-v6.8.11.orig/src/LocalPath.cpp	2018-08-18 16:36:49.000000000 +0200
+++ xcsoar-v6.8.11/src/LocalPath.cpp	2019-05-15 19:56:54.201541488 +0200
@@ -407,6 +407,10 @@
 static TCHAR *
 FindDataPath()
 {
+#ifdef FIXED_DATADIR
+    return DuplicateString(_T(FIXED_DATADIR));
+#else
+
   if (IsAltair() && IsEmbedded()) {
     /* if XCSoarData exists on USB drive, use that, because the
        internal memory is extremely small */
@@ -508,6 +512,7 @@
   }
 
   return nullptr;
+#endif
 }
 
 void
