diff -Naur xcsoar-v6.8.12.orig/build/sdl.mk xcsoar-v6.8.12/build/sdl.mk
--- xcsoar-v6.8.12.orig/build/sdl.mk	2019-07-04 11:11:10.562447130 +0200
+++ xcsoar-v6.8.12/build/sdl.mk	2019-07-04 11:17:23.096055594 +0200
@@ -33,6 +33,10 @@
 
 ifeq ($(USE_SDL2),y)
 $(eval $(call pkg-config-library,SDL,sdl2))
+USE_SDL2_BATTERY ?= y
+ifeq ($(USE_SDL2_BATTERY),y)
+SDL_CPPFLAGS += -DENABLE_SDL_BATTERY
+endif
 else
 $(eval $(call pkg-config-library,SDL,sdl))
 endif
diff -Naur xcsoar-v6.8.12.orig/src/Hardware/Battery.cpp xcsoar-v6.8.12/src/Hardware/Battery.cpp
--- xcsoar-v6.8.12.orig/src/Hardware/Battery.cpp	2019-04-30 22:09:20.000000000 +0200
+++ xcsoar-v6.8.12/src/Hardware/Battery.cpp	2019-07-04 11:15:57.485849513 +0200
@@ -187,7 +187,7 @@
 
 #endif
 
-#if defined(ENABLE_SDL) && (SDL_MAJOR_VERSION >= 2)
+#if defined(ENABLE_SDL_BATTERY)
 
 #include <SDL_power.h>
 
diff -Naur xcsoar-v6.8.12.orig/src/Hardware/Battery.hpp xcsoar-v6.8.12/src/Hardware/Battery.hpp
--- xcsoar-v6.8.12.orig/src/Hardware/Battery.hpp	2019-04-30 22:09:20.000000000 +0200
+++ xcsoar-v6.8.12/src/Hardware/Battery.hpp	2019-07-04 11:16:29.374434121 +0200
@@ -24,11 +24,7 @@
 #ifndef XCSOAR_HARDWARE_BATTERY_H
 #define XCSOAR_HARDWARE_BATTERY_H
 
-#ifdef ENABLE_SDL
-#include <SDL_version.h>
-#endif
-
-#if defined(ANDROID) || (defined(_WIN32_WCE) && !defined(GNAV)) || defined(KOBO) || (defined(ENABLE_SDL) && (SDL_MAJOR_VERSION >= 2))
+#if defined(ANDROID) || (defined(_WIN32_WCE) && !defined(GNAV)) || defined(KOBO) || defined(ENABLE_SDL_BATTERY)
 #define HAVE_BATTERY
 
 #include <stdbool.h>
