diff -Naur linux-4.19.31-ti-r18.orig/tools/iio/iio_generic_buffer.c linux-4.19.31-ti-r18/tools/iio/iio_generic_buffer.c
--- linux-4.19.31-ti-r18.orig/tools/iio/iio_generic_buffer.c	2019-04-10 23:46:55.000000000 +0200
+++ linux-4.19.31-ti-r18/tools/iio/iio_generic_buffer.c	2019-05-31 12:22:48.323000496 +0200
@@ -83,9 +83,9 @@
 	if (info->is_signed) {
 		int8_t val = (int8_t)(input << (8 - info->bits_used)) >>
 			     (8 - info->bits_used);
-		printf("%05f ", ((float)val + info->offset) * info->scale);
+		printf("%05f ", ((float)val * info->scale) + info->offset);
 	} else {
-		printf("%05f ", ((float)input + info->offset) * info->scale);
+		printf("%05f ", ((float)input * info->scale) + info->offset);
 	}
 }
 
@@ -106,9 +106,9 @@
 	if (info->is_signed) {
 		int16_t val = (int16_t)(input << (16 - info->bits_used)) >>
 			      (16 - info->bits_used);
-		printf("%05f ", ((float)val + info->offset) * info->scale);
+		printf("%05f ", ((float)val * info->scale) + info->offset);
 	} else {
-		printf("%05f ", ((float)input + info->offset) * info->scale);
+		printf("%05f ", ((float)input * info->scale) + info->offset);
 	}
 }
 
@@ -129,9 +129,9 @@
 	if (info->is_signed) {
 		int32_t val = (int32_t)(input << (32 - info->bits_used)) >>
 			      (32 - info->bits_used);
-		printf("%05f ", ((float)val + info->offset) * info->scale);
+		printf("%05f ", ((float)val * info->scale) + info->offset);
 	} else {
-		printf("%05f ", ((float)input + info->offset) * info->scale);
+		printf("%05f ", ((float)input * info->scale) + info->offset);
 	}
 }
 
@@ -157,9 +157,9 @@
 			printf("%" PRId64 " ", val);
 		else
 			printf("%05f ",
-			       ((float)val + info->offset) * info->scale);
+			       ((float)val * info->scale) + info->offset);
 	} else {
-		printf("%05f ", ((float)input + info->offset) * info->scale);
+		printf("%05f ", ((float)input * info->scale) + info->offset);
 	}
 }
 
