diff -Naur DirectFB-1.7.7.orig/configure.in DirectFB-1.7.7/configure.in
--- DirectFB-1.7.7.orig/configure.in	2015-02-09 18:16:46.000000000 +0100
+++ DirectFB-1.7.7/configure.in	2019-07-01 11:07:11.935341103 +0200
@@ -2459,7 +2459,7 @@
 
 enable_tslib=no
 if test "$checkfor_tslib" = "yes"; then
-  PKG_CHECK_MODULES([TSLIB], [tslib-1.0 >= 1.0.0], [enable_tslib=yes], [enable_tslib=no])
+  PKG_CHECK_MODULES([TSLIB], [tslib >= 1.0.0], [enable_tslib=yes], [enable_tslib=no])
   if test "$enable_tslib" = "no"; then
      PKG_CHECK_MODULES([TSLIB], [tslib-0.0], [enable_tslib=yes], [enable_tslib=no
        AC_MSG_WARN([*** no tslib -- tslib driver will not be built.])])
