diff -Naur DirectFB-1.7.7.orig/inputdrivers/linux_input/linux_input.c DirectFB-1.7.7/inputdrivers/linux_input/linux_input.c
--- DirectFB-1.7.7.orig/inputdrivers/linux_input/linux_input.c	2014-07-15 08:54:58.000000000 +0200
+++ DirectFB-1.7.7/inputdrivers/linux_input/linux_input.c	2019-07-02 00:40:16.241022116 +0200
@@ -1087,6 +1087,10 @@
                if (test_bit( i, keybit ))
                     num_ext_keys++;
 
+          for (i=KEY_F1; i<=KEY_F10; i++)
+               if (test_bit( i, keybit ))
+                    num_ext_keys++;
+
           for (i=BTN_MOUSE; i<BTN_JOYSTICK; i++)
                if (test_bit( i, keybit ))
                     num_buttons++;
